networking/libsnnet/docker_plugin.go

```diff
@@ -181,12 +197,66 @@ type DockerEpMap struct {
 	m map[string]*DockerEpVal //index: Docker End Point UUID
 }
 
+//NewTable creates a new map
+func (d *DockerEpMap) NewTable() {
+	d.m = make(map[string]*DockerEpVal)
+}
+
+//Name provides the name of the map
+func (d *DockerEpMap) Name() string {
+	return tableNetworkMap
+}
+
+//NewElement allocates and returns an endpoint value
+func (d *DockerEpMap) NewElement() interface{} {
+	return &DockerEpVal{}
+}
+
+//Add adds a value to the map with the specified key
+func (d *DockerEpMap) Add(k string, v interface{}) error {
+	switch val := v.(type) {
```

Comment:
You could use a checked type assertion here instead, which would save you one line of code

```
val, ok := v.(*DockerEpVal)
if !ok {
    return fmt.Errorf("Invalid value type %t", v)
}
d.m[k] = val